src/System.Text.Json/tests/Serialization/Object.ReadTests.cs

```diff
@@ -238,5 +238,53 @@ public static void ReadObject_PublicIndexer()
             Assert.Equal("Value", indexer.NonIndexerProp);
             Assert.Equal(-1, indexer[0]);
         }
+
+        [ActiveIssue(38414)]
+        [Fact]
+        public static void ReadStructWithSimpleClassTest() {
+            SampleStructWithClass testObject = new SampleStructWithClass();
+            testObject.Initialize();
+
+            string json = JsonSerializer.ToString(testObject, testObject.GetType());
+            SampleStructWithClass obj = JsonSerializer.Parse<SampleStructWithClass>(json);
+            obj.Verify();
+        }
+
+        [ActiveIssue(38490)]
+        [Fact]
+        public static void ReadStructObjectValueTest()
+        {
+            SimpleTestStruct testObject = new SimpleTestStruct();
+            testObject.Initialize();
+            testObject.MySampleTestClass = new SimpleTestClass  { MyString = "Hello", MyDouble = 3.14 } ;
+
+            string json = JsonSerializer.ToString(testObject);
+            SimpleTestStruct parsedObject = JsonSerializer.Parse<SimpleTestStruct>(json);
+            parsedObject.Verify();            
+        }
+
+        [ActiveIssue(38435)]
+        [Fact]
+        public static void ReadStructObjectWithSampleValueTest()
+        {
+            string json = "{\"MyInt16\":1,\"MyInt32\":2,\"MyInt64\":3,\"MyUInt16\":4,\"MyUInt32\":5,\"MyUInt64\":6,\"MyByte\":7,\"MySByte\":8,\"MyChar\":\"a\",\"MyString\":\"Hello\",\"MyDecimal\":3.3,\"MyBooleanTrue\":true," +
```

Comment:
Is it possible to re-use `SimpleTestStruct.s_json` here (like the other tests) instead of creating a new string or is this json string intentionally different than the existing one to uncover the issue you found (https://github.com/dotnet/corefx/issues/38435)?

If the issue is reproducible with `SimpleTestStruct.s_json`, we should just use that instead.

Reply:
Yeah that would be better, yes it is the same json as in the issue but unfortunately I was not able to repro the json file again and the same property/data on the issue was not causing this Exception (produced json seems similar but somehow not same exception result). But parsing the json directly causing the same Exception, maybe just delete this test case, anyways there is issue for tracking ...